### DISCUSSION TITLE
Android - Pose detection to animate the 3d model

### DISCUSSION THREAD

**Discussion by vk200000000** [google/filament#7291] [General]
Its not a bug, however, i was able to successfully load the 3d rigged model (GLB) into the modelViewer and also run the default  available animation. Now im stuck in. attaching the pose detected through camera to the 3d model bones to animate. Example like motion capture. Looking for some sample code to get me started on this using this library. This i for Android Kotlin.

**Comment by swafe** [comment]
I am at the same point. Have You found a solution?
At the moment I try to get the specific bone and move it to the specific point. But I am not sure if this is good practice.

```
val tm = modelViewer.engine.transformManager
modelViewer.asset?.getFirstEntityByName(bone_name)?.let { entity ->
                val boneTransformComponent = tm.getInstance(entity)
                val transform = translation(pointxyz)
                tm.setTransform(boneTransformComponent, transpose(transform).toFloatArray())
                modelViewer.animator?.updateBoneMatrices()
}
```
